packages/browser/src/login/oidc/redirectHandler/AuthCodeRedirectHandler.spec.ts

```diff
@@ -553,42 +545,60 @@ describe("AuthCodeRedirectHandler", () => {
     });
 
     it("returns null for the expiration time if none was provided", async () => {
-      mockOidcClient();
-      const mockedOidcClient = jest.requireMock(
-        "@inrupt/oidc-client-ext"
-      ) as any;
-      jest.spyOn(mockedOidcClient, "getBearerToken").mockReturnValueOnce({
-        accessToken: mockAccessTokenBearer(),
-        idToken: mockIdToken(),
-        webId: mockWebId(),
-        // no expiresIn
+      const mockedOidcClient = mockOidcClient();
+      window.fetch = jest.fn();
+      mockLocalStorage({
+        "tmp-resource-server-session-enabled": "false",
       });
+      mockedOidcClient.getBearerToken = jest
+        .fn(getBearerToken)
+        .mockResolvedValueOnce({
+          accessToken: mockAccessTokenBearer(),
+          idToken: mockIdToken(),
+          webId: mockWebId(),
+          // no expiresIn
+        });
 
-      const testIssuer = "some test Issuer";
-      const mockedStorage = mockStorageUtility({
-        "solidClientAuthenticationUser:mySession": {
-          issuer: testIssuer,
-        },
-        "solidClientAuthenticationUser:oauth2_state_value": {
-          sessionId: "mySession",
-        },
-      });
+      const mockedStorage = mockDefaultStorageUtility({ dpop: false });
 
       const authCodeRedirectHandler = getAuthCodeRedirectHandler({
         storageUtility: mockedStorage,
       });
 
       const sessionInfo = await authCodeRedirectHandler.handle(
-        "https://coolsite.com/?code=someCode&state=oauth2_state_value"
+        mockRedirectUrl()
       );
 
       expect(sessionInfo.expirationDate).toBeNull();
     });
+
+    it("clears the oidc-client specific session information", async () => {
+      mockOidcClient();
+      mockLocalStorage({
+        // This mocks how oidc-client stores session information
+        "oidc.oauth2StateValue": "some arbitrary value",
+      });
+      const mockedFetch = jest.fn(global.fetch).mockReturnValue(
```

Comment:
Can you explain the mocked fetch here, is this due to solid-oidc-ext?

Reply:
If `window.fetch` isn't mocked, ```
 const authFetch = await buildAuthenticatedFetch(fetch, tokens.accessToken, {
      dpopKey: tokens.dpopKey,
      refreshOptions,
      eventEmitter,
      expiresIn: tokens.expiresIn,
    });
```
throws. I cleaned up the mock itself though.

Reply:
👍